### PR TITLE
Geth Admin management API's implemented (web3-eth-admin)

### LINKED ISSUE #2637: Add all management API methods of Geth
### Description 

Implement all methods which are listed [here](https://github.com/ethereum/go-ethereum/wiki/Management-APIs).

#### Expected behavior

I can, for example, call the debug methods.

#### Actual behavior

Not all methods of the management API are supported.

### Versions 
* web3.js: latest

### PR DESCRIPTION
**Pull request by princesinha19**
## Description
I implemented all admin management API's provided by geth. 
https://github.com/ethereum/go-ethereum/wiki/Management-APIs#list-of-management-apis

In this PR some of the the code for type is missing. So, I will update it further.

Fixes #2637 Partially

## Type of change
- [x] New feature 
- [x] Enhancement

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no warnings.
- [x] I have updated or added types for all modules I've changed
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran ```npm run test``` in the root folder with success and extended the tests if necessary.
- [ ] I ran ```npm run build``` in the root folder and tested it in the browser and with node.
- [ ] I ran ```npm run dtslint``` in the root folder and tested that all my types are correct
- [ ] I have tested my code on an ethereum test network.

### CODE REVIEW

File: packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * This method will be executed after the RPC request.
+     *
+     * @method afterExecution
+     *
+     * @param {Object} response
+     *
+     * @returns {Array}
+     */
+    afterExecution(response) {
```
Comment (nivida) [inline packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js]: The method ``afterExecution`` has only to be implemented if you would like to format the response. :)

File: packages/web3-eth-admin/src/Admin.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+ * @file index.js
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+
+import {AbstractWeb3Module} from 'web3-core';
+
+export default class Admin extends AbstractWeb3Module {
+    /**
+     * TODO: Add missing documentation for getAccounts, lockAccount and sendTransaction!
```
Comment (nivida) [inline packages/web3-eth-admin/src/Admin.js]: I think this TODO shouldn't be here :)

**Review by nivida — CHANGES_REQUESTED** [review]
Could you extend the documentation for this module? :)

File: packages/web3-eth-admin/src/Admin.js (hunk 2 above)
Comment (princesinha19) [reply packages/web3-eth-admin/src/Admin.js]: Yes right.

File: packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js (hunk 1 above)
Comment (princesinha19) [reply packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js]: Yes got it. Done @nivida

**Comment by princesinha19** [comment]
Yes sure @nivida. I will start documenting it.
Should I add document in this PR or separate PR?

**Comment by nivida** [comment]
I would add the docs directly here to this PR. Thanks! :)

File: packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+*/
+/**
+ * @file AdminAddPeerMethod.js
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+
+import AbstractMethod from '../../../lib/methods/AbstractMethod';
+
+export default class AdminAddPeerMethod extends AbstractMethod {
```
Comment (nivida) [inline packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js]: I would call this method ``AddPeerMethod`` because it's already in the ``admin`` folder :)

File: packages/web3-core-method/src/methods/admin/AdminStartRpcMethod.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+*/
+/**
+ * @file AdminStartRpcMethod.js
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+
+import AbstractMethod from '../../../lib/methods/AbstractMethod';
+
+export default class AdminStartRpcMethod extends AbstractMethod {
```
Comment (nivida) [inline packages/web3-core-method/src/methods/admin/AdminStartRpcMethod.js]: Same here :)

File: packages/web3-core-method/src/methods/admin/GetAdminDataDirectoryMethod.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+*/
+/**
+ * @file GetAdminDataDirectoryMethod.js
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+
+import AbstractMethod from '../../../lib/methods/AbstractMethod';
+
+export default class GetAdminDataDirectoryMethod extends AbstractMethod {
```
Comment (nivida) [inline packages/web3-core-method/src/methods/admin/GetAdminDataDirectoryMethod.js]: I would remove the ``Get`` prefix here and rename it to ``AdminDataDirectoryMethod``.

File: packages/web3-core-method/src/methods/admin/GetAdminNodeInfoMethod.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+*/
+/**
+ * @file GetAdminNodeInfoMethod.js
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+
+import AbstractMethod from '../../../lib/methods/AbstractMethod';
+
+export default class GetAdminNodeInfoMethod extends AbstractMethod {
```
Comment (nivida) [inline packages/web3-core-method/src/methods/admin/GetAdminNodeInfoMethod.js]: ``AdminNodeInfoMethod``

File: packages/web3-core-method/src/methods/admin/SetAdminSolcMethod.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+*/
+/**
+ * @file SetAdminSolcMethod.js
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+
+import AbstractMethod from '../../../lib/methods/AbstractMethod';
+
+export default class SetAdminSolcMethod extends AbstractMethod {
```
Comment (nivida) [inline packages/web3-core-method/src/methods/admin/SetAdminSolcMethod.js]: ``SetSolcMethod``

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/web3-core-method/src/methods/admin/GetAdminDataDirectoryMethod.js (hunk 5 above)
Comment (princesinha19) [reply packages/web3-core-method/src/methods/admin/GetAdminDataDirectoryMethod.js]: Done

File: packages/web3-core-method/src/methods/admin/AdminStartRpcMethod.js (hunk 4 above)
Comment (princesinha19) [reply packages/web3-core-method/src/methods/admin/AdminStartRpcMethod.js]: Done

File: packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js (hunk 3 above)
Comment (princesinha19) [reply packages/web3-core-method/src/methods/admin/AdminAddPeerMethod.js]: Done.

File: packages/web3-core-method/src/methods/admin/GetAdminNodeInfoMethod.js (hunk 6 above)
Comment (princesinha19) [reply packages/web3-core-method/src/methods/admin/GetAdminNodeInfoMethod.js]: Done.

File: packages/web3-core-method/src/methods/admin/SetAdminSolcMethod.js (hunk 7 above)
Comment (princesinha19) [reply packages/web3-core-method/src/methods/admin/SetAdminSolcMethod.js]: Done.

**Comment by nivida** [comment]
@princesinha19 Sorry if I annoy you but could you also update the other method object names in the same pattern as the already changed methods?:)

**Comment by princesinha19** [comment]
@nivide no problem. I will change it. Can you point me to the methods? 
I changed setSolcMethod, StartRPCMethod, StopRPCMethod, StartWSMethod, StopWSMethod, AddPeerMethod, GetAdminPeerMethod, AdminNodeInfoMethod, AdminDataDirectoryMethod.

should I change GetAdminPeerMethod to GetPeerMethod?

**Comment by nivida** [comment]
Yes, I would change the name of these methods: 
``AdminDataDirectoryMethod, AdminNodeInfoMethod, GetAdminPeersMethod``

To: 
``DataDirectoryMethod, NodeInfoMethod, PeersMethod``

**Review by nivida — APPROVED** [review]

File: packages/web3-eth-admin/types/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    getPeers(
+        callback?: (error: Error, result: []) => void
+    ): Promise<[]>;
+
+    setSolc(
+        path: string,
+        callback?: (error: Error, result: string) => void
+    ): Promise<string>;
+
+    startRPC(
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts:50]: Having 4 optional parameter in these typings makes you have to write horrible ts code. I would prefer if you did every outcome and overloaded the types:

```
startRPC(host: string): Promise<boolean>
startRPC(port: Hex): Promise<boolean>
startRPC(cors: string): Promise<boolean>
startRPC(apis: string): Promise<boolean>
startRPC(host: string, port: string): Promise<boolean>
... and so on with all 4 of the outcomes.
```

This will make this a lot nicer to work with.

File: packages/web3-eth-admin/types/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        host?: string,
+        port?: Hex,
+        cors?: string,
+        apis?: string,
+        callback?: (error: Error, result: boolean) => void
+    ): Promise<boolean>;
+
+    startWS(
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts:58]: same as the above ^

File: packages/web3-eth-admin/types/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    getadminDataDirectory(
+        callback?: (error: Error, result: string) => void
+    ): Promise<string>;
+
+    getNodeInfo(
+        callback?: (error: Error, result: NodeInfo) => void
+    ): Promise<NodeInfo>;
+
+    getPeers(
+        callback?: (error: Error, result: []) => void
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: please put `result: any[]` unless you know the type which can be in that array.

File: packages/web3-eth-admin/types/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+        host?: string,
+        port?: Hex,
+        cors?: string,
+        apis?: string,
+        callback?: (error: Error, result: boolean) => void
+    ): Promise<boolean>;
+
+    startWS(
+        host?: string,
+        port?: Hex,
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: why make port accept a hex? it should always be a number i reckon and the controller handles it to a hex if it needs to be like that for the JSONRPC method. This should really just be `port: number` to make the interface nice.

File: packages/web3-eth-admin/types/index.d.ts [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    ): Promise<[]>;
+
+    setSolc(
+        path: string,
+        callback?: (error: Error, result: string) => void
+    ): Promise<string>;
+
+    startRPC(
+        host?: string,
+        port?: Hex,
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: why make port accept a hex? it should always be a number i reckon and the controller handles it to a hex if it needs to be like that for the JSONRPC method. This should really just be `port: number` to make the interface nice.

File: packages/web3-eth-admin/types/tests/admin-tests.ts [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+admin.setSolc('/usr/bin/solc');
+
+// $ExpectType Promise<string>
+admin.setSolc(
+    '/usr/bin/solc',
+    (error: Error, result: string) => {}
+)
+
+// $ExpectType Promise<boolean>
+admin.startRPC();
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/tests/admin-tests.ts:67]: You have not tested every single possibility for `startRPC`, please can you.

File: packages/web3-eth-admin/types/tests/admin-tests.ts [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+admin.startRPC("127.0.0.1", 8545, "", "eth,net,web3");
+
+// $ExpectType Promise<boolean>
+admin.startRPC(
+    "127.0.0.1", 8545, "", "eth,net,web3",
+    (error: Error, result: boolean) => {}
+);
+
+// $ExpectType Promise<boolean>
+admin.startWS();
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/tests/admin-tests.ts:166]: You have not tested every single possibility for `startWS`, please can you.

**Comment by joshstevens19** [comment]
If this has to get to through web3.eth you have to add this to the Eth typings class or it wont be exposed in the global scope when ts users try to use it.

https://github.com/ethereum/web3.js/blob/1.0/packages/web3-eth/types/index.d.ts

Please add it

File: packages/web3-eth-admin/types/index.d.ts [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+export class Admin extends AbstractWeb3Module {
+    constructor(provider: provider, net?: net.Socket|null, options?: Web3ModuleOptions, accounts?: Accounts|null);
+
+    addPeer(
+        url: string,
+        callback?: (error: Error, result: boolean) => void
+    ): Promise<boolean>;
+
+    getadminDataDirectory(
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: In your docs you state this as `getDataDirectory` please change the typing's to match.

File: packages/web3-eth-admin/types/index.d.ts [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+        callback?: (error: Error, result: string) => void
+    ): Promise<string>;
+
+    getNodeInfo(
+        callback?: (error: Error, result: NodeInfo) => void
+    ): Promise<NodeInfo>;
+
+    getPeers(
+        callback?: (error: Error, result: []) => void
+    ): Promise<[]>;
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: can you please type this interface, seen your docs and you can create a new interface for this.

File: packages/web3-eth-admin/types/index.d.ts [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+    stopRPC(
+        callback?: (error: Error, result: boolean) => void
+    ): Promise<boolean>;
+
+    stopWS(
+        callback?: (error: Error, result: boolean) => void
+    ): Promise<boolean>;
+}
+
+export type Hex = string | number;
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: if your going to reuse this `hex` all the time like you have please put it in `web3-core`, personally i think just do the type as `string | number`. I think its overkill creating a custom type and the rest of lib except for utils doesn't do it like this. Just my view :+1:

File: packages/web3-eth-admin/types/index.d.ts (hunk 12 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: In Geth document, it is specified that port can be `rpc.HexNumber`. That's why i used Hex (string | number). https://github.com/ethereum/go-ethereum/wiki/Management-APIs#admin_startrpc

File: packages/web3-eth-admin/types/tests/admin-tests.ts (hunk 13 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/tests/admin-tests.ts:67]: yes, I will.

File: packages/web3-eth-admin/types/tests/admin-tests.ts (hunk 14 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/tests/admin-tests.ts:166]: cool, I will improve test.

File: packages/web3-eth-admin/types/index.d.ts (hunk 15 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: yes, I missed it. thanks.

File: packages/web3-eth-admin/types/index.d.ts (hunk 16 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: yes sure.

File: packages/web3-eth-admin/types/index.d.ts (hunk 17 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: cool. sounds good.

File: packages/web3-eth-admin/types/index.d.ts (hunk 10 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: I will check and create an interface.

File: packages/web3-eth-admin/types/index.d.ts (hunk 12 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: okay but I will change it to the number.

**Comment by nivida** [comment]
> If this has to get to through web3.eth you have to add this to the Eth typings class or it wont be exposed in the global scope when ts users try to use it.

The eth management modules should not get added to the web3.eth module this would blow up the bundle size for "normal" DApp's. :)

**Comment by princesinha19** [comment]
> The eth management modules should not get added to the web3.eth module this would blow up the bundle size for "normal" DApp's. :)

Right.

@nivida I can see. all the methods of web3-eth-personal, web3-eth-accounts are added in **Eth typings class** of web3-eth.

File: packages/web3-eth-admin/types/index.d.ts (hunk 8 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts:50]: I didn't see any side effect of the above code. @joshstevens19.
If I will do a method overloading then, I have to write many cases. That will become horrible. Can you explain a bit?

Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts:50]: Done, I overloaded the types.

**Comment by princesinha19** [comment]
I overloaded the types and also did all the requested changes.

After overloading the method some error got occurred in build pipeline. Could please take a look at build pipeline @joshstevens19.

File: packages/web3-eth-admin/types/index.d.ts [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+    getPeers(
+        callback?: (error: Error, result: any[]) => void
+    ): Promise<any[]>;
+
+    setSolc(
+        path: string,
+        callback?: (error: Error, result: string) => void
+    ): Promise<string>;
+
+    startRPC(callback?: (error: Error, result: boolean) => void): Promise<boolean>
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: These types don't go well due to 4 optional parameters which makes it impossible to make this nice typing's. If you repeat stuff like 

```
startRPC(host: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
startRPC(cors: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
```

in TS it wont know which type to resolve to if you await them as both accept a string. 

Sorry to be a pain please revert back to what you had before as this wont work at all. I think 1 type with 4 optional parameters is the only way here unless you broke the methods down which i thinks overkill. You can also revert my tslint change and remove the `"unified-signatures": false` line.

What you did on the testing will still need to test all the situations but put `null` in the parameters if you need to say test parameter 1 and 3. Optional functions like this especially 4 make typing's not very nice.

File: packages/web3-eth-admin/types/index.d.ts [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+    startRPC(port: number, cors: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+    startRPC(port: number, apis: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+    startRPC(cors: number, apis: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+    startRPC(host: string, port: number, cors: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+    startRPC(host: string, port: number, apis: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+    startRPC(host: string, cors: string, apis: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+    startRPC(port: number, cors: string, apis: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+    startRPC(host: string, port: number, cors: string, apis: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+
+    startWS(callback?: (error: Error, result: boolean) => void): Promise<boolean>
```
Comment (joshstevens19) [inline packages/web3-eth-admin/types/index.d.ts]: These types don't go well due to 4 optional parameters which makes it impossible to make this nice typing's. If you repeat stuff like 

```
startWS(host: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
startWS(cors: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
```

in TS it wont know which type to resolve to if you await them as both accept a string. 

Sorry to be a pain please revert back to what you had before as this wont work at all. I think 1 type with 4 optional parameters is the only way here unless you broke the methods down which i thinks overkill. You can also revert my tslint change and remove the `"unified-signatures": false` line.

What you did on the testing will still need to test all the situations but put `null` in the parameters if you need to say test parameter 1 and 3. Optional functions like this especially 4 make typing's not very nice.

**Comment by joshstevens19** [comment]
@nivida @princesinha19 If this is a standalone module then great but in regards to what you said about `web3-eth-personal, web3-eth-accounts` if you read my comment here it will make sense.

https://github.com/ethereum/web3.js/pull/2660#issuecomment-482510665

I think all your PRs are standalone modules the user should only be able to get to it if they import the module and not through the public interface. If thats the case then great. We can't and should not remove the `web3-eth-personal, web3-eth-accounts` dependency as in the public API docs it states you can get to it through `web.eth` interface. Which i know a lot of people do. 

https://web3js.readthedocs.io/en/1.0/web3-eth.html#personal
https://web3js.readthedocs.io/en/1.0/web3-eth.html#accounts

:+1:

File: packages/web3-eth-admin/types/index.d.ts (hunk 18 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: okay so, I will revert it as it was before @joshstevens19

**Comment by princesinha19** [comment]
I am okay with adding it to web3.eth too. But, I agree with @nivida that it will unnecessarily increase the bundle size. So, I think we can separate the management API's and will keep it standalone modules. 
what's your thinking @nivida & @joshstevens19?

**Comment by joshstevens19** [comment]
Yeah i agree with that, keeping it separate! @princesinha19 :+1: Also i forgot to say brilliant work on the PRs done 👍

File: packages/web3-eth-admin/types/index.d.ts (hunk 18 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: But If I will revert it to as it was before then, I will not able to test all the outcomes @joshstevens19.

```
    startRPC(
        host?: string,
        port?: number,
        cors?: string,
        apis?: string,
        callback?: (error: Error, result: boolean) => void
    ): Promise<boolean>;
```
consider this test:

```
// $ExpectType Promise<boolean>
admin.startRPC(8545);
```
We will not able to test this.

File: packages/web3-eth-admin/types/index.d.ts (hunk 19 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: But If I will revert it to as it was before then, I will not able to test all the outcomes @joshstevens19.

```
    startWS(
        host?: string,
        port?: number,
        cors?: string,
        apis?: string,
        callback?: (error: Error, result: boolean) => void
    ): Promise<boolean>;
```
consider this test:

```
// $ExpectType Promise<boolean>
admin.startWS(8546);
```
We will not able to test this.

Comment (joshstevens19) [reply packages/web3-eth-admin/types/index.d.ts]: yes hence why i said put `null` in the parameters you need to ignore. They are all optional parameters so we need to cover each case really i.e.

```
// $ExpectType Promise<boolean>
admin.startWS(null, 8546);
```

File: packages/web3-eth-admin/types/index.d.ts (hunk 18 above)
Comment (joshstevens19) [reply packages/web3-eth-admin/types/index.d.ts]: yes hence why i said put `null` in the parameters you need to ignore. They are all optional parameters so we need to cover each case really i.e.

```
// $ExpectType Promise<boolean>
admin.startRPC(null, 8546);
```

File: packages/web3-eth-admin/types/index.d.ts (hunk 19 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: we can't assign **null** parameter.

File: packages/web3-eth-admin/types/index.d.ts (hunk 18 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: Argument of type 'null' is not assignable to parameter of type 'string | undefined'.ts(2345).

Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: For that , I need to add "strictNullChecks": false.

File: packages/web3-eth-admin/types/index.d.ts (hunk 19 above)
Comment (princesinha19) [reply packages/web3-eth-admin/types/index.d.ts]: Please review. I did this change.

**Review by joshstevens19 — APPROVED** [review]

File: packages/web3-eth-admin/rollup.config.js [hunk 20]
```diff
@@ -0,0 +1,4 @@
+import pkg from './package.json';
+import rollupConfig from '../../rollup.config';
+
+export default rollupConfig('Web3Shh', pkg.name);
```
Comment (nivida) [inline packages/web3-eth-admin/rollup.config.js]: ``Web3Shh`` should be ``Web3Admin`` :)

File: packages/web3-eth-admin/src/Admin.js [hunk 21]
```diff
@@ -0,0 +1,219 @@
...
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.js
```
Comment (nivida) [inline packages/web3-eth-admin/src/Admin.js]: ``index.js`` should be ``Admin.js``

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/web3-eth-admin/src/Admin.js (hunk 21 above)
Comment (nivida) [reply packages/web3-eth-admin/src/Admin.js]: I'll merge it after these small changes.

File: packages/web3-eth-admin/rollup.config.js (hunk 20 above)
Comment (princesinha19) [reply packages/web3-eth-admin/rollup.config.js]: Yeah, sorry I missed this.

**Review by nivida — APPROVED** [review]
